src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+            dy_np1=F(t_np1, y_np1)
+
+            #---################################################################
+            # Correction-Evaluaion (CE) Loop for implicit solver
+            #     See Hairer p. 360 for a general description of this method
+            #---################################################################
+
+            # Set CE_loops to be how many correction-evaulation cycles
+            # desired. As least one loop is necessary, for the sake of the
+            # adaptive step selection
+            CE_loops = 1
+            if n>=steporder+1
+                for loop_counter=1:CE_loops
+                    # (C)orrection of y_np1
+                    # Add new coefficients  based on this prediction
+                    # computation of ϕ, ϕstar using recurrence relations
+                    ϕ_coefs!(ϕ, ϕstar, dy_np1, 1:steporder+1)
+
+                    # Calculation of g_{steporder + 1}(n) coefficients using
+                    # recurrence relations has already been done above, and does
+                    # not depend on dy_np1 or y_np1, so no update necessary
+
+                    # We correct the computation of y_np1 using y_np1
+                    # as part of our prediction Hairer equation 5.7
+                    y_np1 += dt * g[steporder+1] * ϕ[3, steporder+1]
+
+                    # (E)valuate estimate for next derivative again
+                    dy_np1=F(t_np1, y_np1)
+                end
+            end
+            #---################################################################
+            # Error Check for step size `dt` and Variable next step selection:
+            #     Check error to determine if step size should be accepted:
+            #     ->if yes, compute `next_dt` for next step
+            #     ->if no, use this `next_dt`` for next attempt at this step
+            #---################################################################
+
+            #Don't allow increases of steps until n >= steporder + 2
+            if n==1
+                successful_step = true
+                next_dt = dt / 4
+            elseif n<steporder+2
+                yerr_k = dt * (g[steporder+1] - g[steporder]) * ϕ[3,steporder+1]
+                err_k, next_dt,timeout_VS = stepsize_hw92!(dt, tdir, y[n], y_np1, yerr_k, steporder,
+                                timeout_VS, dof, abstol, reltol, maxstep, norm)
+                if err_k <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    dt = fac * next_dt
+                end
+            else
+                # We calculate local truncation error, LE_{k+1}; the difference
+                # between implicit results for order k and k+1.
+
+                # We need ϕ_{k+1}(n+1) and g_{k+1}(n);the latter has already
+                # been calculated above, the former still needs calculation
+                ϕ_coefs!(ϕ, ϕstar, dy_np1, steporder+2)
```

Comment:
Is it correct to call this with a non-range as last argument?  Aren't the `ϕ[3, j]` dependent on `ϕ[3, j-1]`, so all would need to be calculated.